packages/cli/src/ceramic-daemon.ts

```diff
@@ -364,8 +365,28 @@ export class CeramicDaemon {
 
     // This needs to happen after the DID is set up so that any stream operations have a valid DID to use
     if (opts.stateStore?.mode == StateStoreMode.S3) {
+      let networkName = params.networkOptions.name
+
+      // Check if ELP bucket is used
```

Comment:
I think this is generally the right idea, but I think it'd be better done inside of the S3Store's initialization instead of up here.

Also, I know this is a slight scope increase from what we originally talked about, but it looks like the behavior for leveldb just *always* uses "elp" as the storage location for state store data (https://github.com/ceramicnetwork/js-ceramic/blob/c61623e9761872070fa86768137d9286ba54cf2b/packages/core/src/store/level-db-store.ts#L65), instead of only using it if there's existing data there and preferring "mainnet" for new deploys.  Ideally both state store implementations would have the same logic for preferring "mainnet" as the location but using "elp" if data is detected there.

Reply:
I looked into putting the logic in the `S3Store` class directly but it seems all the initialization logic is in the constructor and there is no async initialization method?
So I guess we can either have this async logic out of the store or change the store logic to support async initialization, but that seems like it would add complexity to handle as currently all the stores creation seems synchronous?

I think the alternative for S3 could be to add the ELP check in the `createStore()` method of `S3StoreMap` that is handled as if it was async even though it only runs sync logic? 
I wonder if we need to check the bucket existence for every `createStore()` call though, or running the check only once would be enough to detect the bucket we need to use?

For the LevelDB store I think that can work either way as we can use Node's synchronous FS APIs if needed?

Reply:
I think adding an async init() should be pretty straightforward.  There's already a place we're initializing the state store already (`Repository.init()` calls `pinStore.open()` which calls `stateStore.open()`), it just doesn't get threaded all the way down to the underlying S3Store or LevelDbStore, but it's be easy to plug them in.  Probably can just add a call to store.init() into the open method of the ObjectStore: https://github.com/ceramicnetwork/js-ceramic/blob/c61623e9761872070fa86768137d9286ba54cf2b/packages/core/src/store/object-store.ts#L29

Reply:
OK thanks!
I moved the logic to the S3Store and updated the logic in the LevelDB store.

There are existing tests for the ELP/mainnet change for the LevelDB store so I think it's covered, but any idea how to test the S3Store behavior please? Is the bucket existence check correct?